core/services/relay/evm/types/types_test.go

```diff
@@ -39,3 +45,112 @@ FeedID = "0x%x"
+						ContractPollingFilter: ContractPollingFilter{
+							GenericEventNames: []string{"event1", "event2"},
+							PollingFilter: PollingFilter{
+								Topic2:       evmtypes.HashArray{common.HexToHash("0x1abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+								Topic3:       evmtypes.HashArray{common.HexToHash("0x2abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+								Topic4:       evmtypes.HashArray{common.HexToHash("0x3abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+								Retention:    models.Interval(time.Minute * 1),
+								MaxLogsKept:  100,
+								LogsPerBlock: 10,
+							},
+						},
+						Configs: map[string]*ChainReaderDefinition{
+							"config1": {
+								CacheEnabled:      true,
+								ChainSpecificName: "specificName1",
+								ReadType:          Method,
+								InputModifications: codec.ModifiersConfig{
+									&codec.EpochToTimeModifierConfig{
+										Fields: []string{"ts"},
+									},
+									&codec.RenameModifierConfig{
+										Fields: map[string]string{
+											"a": "b",
+										},
+									},
+								},
+								OutputModifications: codec.ModifiersConfig{
+									&codec.EpochToTimeModifierConfig{
+										Fields: []string{"ts"},
+									},
+									&codec.RenameModifierConfig{
+										Fields: map[string]string{
+											"c": "d",
+										},
+									},
+								},
+								ConfidenceConfirmations: map[string]int{"0.0": 0, "1.0": -1},
+								EventDefinitions: &EventDefinitions{
+									GenericTopicNames:    map[string]string{"TopicKey1": "TopicVal1"},
+									GenericDataWordNames: map[string]uint8{"DataWordKey": 1},
+									InputFields:          []string{"Event1", "Event2"},
+									PollingFilter: &PollingFilter{
+										Topic2:       evmtypes.HashArray{common.HexToHash("0x4abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+										Topic3:       evmtypes.HashArray{common.HexToHash("0x5abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+										Topic4:       evmtypes.HashArray{common.HexToHash("0x6abbe4784b1fb071039bb9cb50b82978fb5d3ab98fb512c032e75786b93e2c52")},
+										Retention:    models.Interval(time.Minute * 1),
+										MaxLogsKept:  100,
+										LogsPerBlock: 10,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var config ChainReaderConfig
+			config.Contracts = make(map[string]ChainContractReader)
+			assert.Nil(t, json.Unmarshal([]byte(tt.jsonInput), &config))
```

Comment:
Maybe require nil, so the following line doesn't make noise and distract from this.